lib/claim/token-holder-revenue-fund-contracts-ensemble.js

```diff
@@ -0,0 +1,359 @@
+                );
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the count of closed accruals for the given currency
+     * @param {Currency} currency - The currency
+     * @returns {Promise} A promise that resolves into the count of closed accruals
+     */
+    async closedAccrualsCount(currency) {
+        const ensemble = _ensemble.get(this);
+
+        const count = (await Promise.all(ensemble.map(c => c.contract.closedAccrualsCount(
+            currency.ct.toString(), currency.id
+        ))))
+            .reduce((a, c) => a.add(c), bigNumberify(0));
+
+        return count;
+    }
+
+    /**
+     * Get claimable amount for the given span of accruals
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @param {number} startAccrual - The lower accrual index boundary
+     * @param {number} endAccrual - The upper accrual index boundary
+     * @returns {Promise} A promise that resolves into a BigNumber value representing the claimable amount.
+     */
+    async claimableAmountByAccruals(wallet, currency, startAccrual, endAccrual) {
+        if (startAccrual > endAccrual)
+            throw new Error(`Ordinality mismatch of startAccrual > endAccrual (${startAccrual} > ${endAccrual})`);
+
+        if (!this.isSpanned(currency))
+            await this.span(currency);
+
+        const ensemble = _ensemble.get(this);
+        const currencyKey = currencyToKey(currency);
+
+        let claimableFees = bigNumberify(0);
+
+        for (const configuration of ensemble) {
+            const span = configuration.spansByCurrency.get(currencyKey);
+
+            if (endAccrual < span.startAccrual || startAccrual > span.endAccrual)
+                continue;
+
+            claimableFees = claimableFees.add(
+                await configuration.contract.claimableAmountByAccruals(
+                    wallet.address, currency.ct.toString(), currency.id,
+                    Math.max(startAccrual, span.startAccrual),
+                    Math.min(endAccrual, span.endAccrual)
+                )
+            );
+        }
+
+        return claimableFees;
+    }
+
+    /**
+     * Claim and stage for the given span of accruals
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @param {number} startAccrual - The lower accrual index boundary
+     * @param {number} endAccrual - The upper accrual index boundary
+     * @param [options] - An optional object containing the parameters for gasLimit and gasPrice
+     * @returns {Promise} A promise that resolves into an array of transaction hashes.
+     */
+    async claimAndStageByAccruals(wallet, currency, startAccrual, endAccrual, options) {
+        if (startAccrual > endAccrual)
+            throw new Error(`Ordinality mismatch of startAccrual > endAccrual (${startAccrual} > ${endAccrual})`);
+
+        if (!this.isSpanned(currency))
```

Comment:
Candidate for removal?

Reply:
Being replaced by `if (!isSpanned.call(this, currency))`.